### PR TITLE
[Client encryption]: Adds JsonNode support on deserialization path

### PR DESCRIPTION
**Pull request by JanHyka**
# Pull Request Template

## Description

- Added support for System.Text.JsonNode DOM on deserializer and decryption path
- Drops custom byte[] converter as JsonNode can take Memory<byte> directly

To be processed after #4780 

## Type of change

Please delete options that are not relevant.

- [] New feature (non-breaking change which adds functionality)
- [] This change requires a documentation update

## Closing issues

Contributes to #4678

### CODE REVIEW

File: Microsoft.Azure.Cosmos.Encryption.Custom/src/EncryptionOptions.cs [hunk 1]
```diff
@@ -6,6 +6,25 @@ namespace Microsoft.Azure.Cosmos.Encryption.Custom
 {
     using System.Collections.Generic;
 
+    /// <summary>
+    /// API for JSON processing
+    /// </summary>
+    public enum JsonProcessor
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos.Encryption.Custom/src/EncryptionOptions.cs:12]: CosmosClientOptions do support two serialization modes (STJ and Newtonsoft). 
Can this be inferred from ClientOptions intead?

File: Microsoft.Azure.Cosmos.Encryption.Custom/tests/Microsoft.Azure.Cosmos.Encryption.Custom.Performance.Tests/Readme.md [hunk 2]
```diff
@@ -1,19 +1,25 @@
...
+| **Encrypt** |               **10** |     **Newtonsoft** |    **83.23 μs** |  **1.608 μs** |  **2.147 μs** |  **0.6104** |  **0.1221** |       **-** |  **166.64 KB** |
+| Decrypt |               10 |     Newtonsoft |   101.62 μs |  1.638 μs |  2.349 μs |  0.6104 |  0.1221 |       - |  152.53 KB |
+| **Encrypt** |               **10** | **SystemTextJson** |    **41.49 μs** |  **0.317 μs** |  **0.464 μs** |  **0.4272** |  **0.0610** |       **-** |  **102.99 KB** |
+| Decrypt |               10 | SystemTextJson |    41.53 μs |  0.505 μs |  0.725 μs |  0.3662 |  0.0610 |       - |   94.09 KB |
+| **Encrypt** |              **100** |     **Newtonsoft** | **1,081.23 μs** | **13.538 μs** | **18.978 μs** | **25.3906** | **23.4375** | **21.4844** | **1638.32 KB** |
+| Decrypt |              100 |     Newtonsoft | 1,135.00 μs | 32.719 μs | 45.867 μs | 17.5781 | 15.6250 | 15.6250 | 1229.52 KB |
+| **Encrypt** |              **100** | **SystemTextJson** |   **819.27 μs** | **22.564 μs** | **33.074 μs** | **25.3906** | **25.3906** | **25.3906** |  **942.76 KB** |
+| Decrypt |              100 | SystemTextJson |   698.30 μs | 20.402 μs | 29.905 μs | 21.4844 | 21.4844 | 21.4844 |  927.92 KB |
```
Comment (kirankumarkolli) [inline Microsoft.Azure.Cosmos.Encryption.Custom/tests/Microsoft.Azure.Cosmos.Encryption.Custom.Performance.Tests/Readme.md:25]: awesome, thank you!

File: Microsoft.Azure.Cosmos.Encryption.Custom/src/EncryptionOptions.cs (hunk 1 above)
Comment (JanHyka) [reply Microsoft.Azure.Cosmos.Encryption.Custom/src/EncryptionOptions.cs:12]: Edit: there is no direct way betwen CosmosClientOptions and EncryptionOptions (in fact Cosmos.Encryption.Custom doesn't seem to reference CosmosClientOptions at all). While probably doable it seems we'd have to change Cosmos.Encryption.Custom api (it passes EncyrptionOptions everywhere). Requesting guidance on how to proceed.

File: Microsoft.Azure.Cosmos.Encryption.Custom/tests/Microsoft.Azure.Cosmos.Encryption.Custom.Performance.Tests/Readme.md (hunk 2 above)
Comment (JanHyka) [reply Microsoft.Azure.Cosmos.Encryption.Custom/tests/Microsoft.Azure.Cosmos.Encryption.Custom.Performance.Tests/Readme.md:25]: Happy to help :-)

**Review by kirankumarkolli — APPROVED** [review]

**Pull request merged**